### DISCUSSION TITLE
Cloud Functions for Bot development

### DISCUSSION THREAD

**Discussion by akhromieiev** [firebase/functions-samples#810] [Q&A] [answered]
Hello! I'm just curious. Could we recognize cloud functions for bot-development? E.g. do bots for Telegram, Viber, Messenger etc?
Most of these platforms require just a Webhook URL. Could we use HTTPs trigger for this purpose?

**Comment by samtstern** [comment, marked as answer]
@akhromieiev absolutely!  I have built many bots using Cloud Functions as webhook handlers including the famous @google-oss-bot (source: https://github.com/firebase/oss-bot) 

All best practices about general webhook development still apply.  HTTP functions are public so you need to make the incoming request is real (GitHub uses a secret token) before handling it.

Also for developing webhooks locally in the Cloud Functions emulator, check out https://smee.io/ which is a great tool.

**Reply by akhromieiev** [reply]
Cool! Thanks!
